iroh-dns-server/src/http/rate_limiting.rs

```diff
@@ -0,0 +1,38 @@
+use std::time::Duration;
+
+use governor::{clock::QuantaInstant, middleware::NoOpMiddleware};
+use tower_governor::{
+    governor::GovernorConfigBuilder, key_extractor::PeerIpKeyExtractor, GovernorLayer,
+};
+
+/// Create the default rate-limiting layer.
+///
+/// This spawns a background thread to clean up the rate limiting cache.
+pub fn create() -> GovernorLayer<'static, PeerIpKeyExtractor, NoOpMiddleware<QuantaInstant>> {
+    // configure rate limiting
```

Comment:
can we record a metric of how much we have limited?

Reply:
Actually, not super easily. Likely will have to downcast some error from https://docs.rs/tower_governor
That crate was the only straightforward rate-limit middleware I could find, however while it is configurable, I'm not yet sure how well it composes for us. Will give it more thought and read its source code.